strongbox-web-core/src/main/java/org/carlspring/strongbox/domain/FileContent.java

```diff
@@ -0,0 +1,61 @@
+package org.carlspring.strongbox.domain;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.nio.file.Path;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import org.apache.commons.io.FileUtils;
+
+public class FileContent 
+{
+
+    private String name;
+
+    private String size;
+
+    private String lastModified;
+    
+    @JsonCreator
+    public FileContent( @JsonProperty("name") String name,
```

Comment:
why you need this `@JsonProperty` here?

Reply:
https://stackoverflow.com/questions/21920367/why-when-a-constructor-is-annotated-with-jsoncreator-its-arguments-must-be-ann

P.S. : This is my first time using JsonCreator.

Reply:
ok, why do you need `JsonCreator` here? :)

Reply:
Getting foll. error without it : 
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.carlspring.strongbox.domain.DirectoryContent` (no Creators, like default construct, exist): cannot deserialize from Object value (no delegate- or property-based Creator)

I think since we do not have a kind of default constructor for our POJO, the object mapper cannot create the JSON object from the String as it does not know how to. So this constructor points  the deserializer to this JsonCreator for making JSON object

Reply:
> P.S.: this error occurred in Junit Test where I need to deserialize the string to object.

oh, now it's clear :)

Reply:
but if you will create noags constructor and add setters then you can remove this annotations from constructor and it will look much more clearer